### PR TITLE
Singularity environment

### PR DESCRIPTION
**Pull request by haz**
This should get us started. Modelled the example files off of [your stuff](https://gist.github.com/gfrances/40689da680e44c103575e7d26e201660), @gfrances . Things look ok?

### CODE REVIEW

File: environments/singularity-builder/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+## To run
+
+```bash
+curl -fsSL https://raw.githubusercontent.com/AI-Planning/planutils/master/environments/singularity-builder/setup.sh | sh
+```
+
+## Note
+
+If using inside of the `planutils` docker imagine, you must run it with `--privileged` mode.
```
Comment (Maumagnaguagno) [inline environments/singularity-builder/README.md:13]: docker imagine => docker image

Comment (haz) [reply environments/singularity-builder/README.md:13]: But are you sure that's really a bug? ;)

**Comment by gfrances** [comment]
Looks good to me, @haz; I wonder though if it is really worth asking for sudo rights only to do the Docker installation (of course we cannot do more than that, since the actual Docker recipe will need to be customized). I usually don't like scripts asking for admin rights :-)

If the goal is to make the installation as easy as possible, perhaps an alternative would be to ship an independent (and easier to inspect) `install-docker.sh` script, and have separate sample Singularity / Docker recipes in the `examples` dir?

File: environments/singularity-builder/setup.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+[ "$UID" -eq 0 ] || (echo "installation requires root access"; exec sudo "$0" "$@")
+
+apt update
+apt install curl
+
+# Install docker
+curl -fsSL https://get.docker.com | sh
```
Comment (gfrances) [inline environments/singularity-builder/setup.sh:9]: Not sure if this could require the `ca-certificates` package?

**Comment by haz** [comment]
It kind of already works like that, putting the examples in their own directory. Only other steps are root-required (apt installation and docker installation).

I've put a guard at the start of the script now to confirm the user wants to proceed. I think that's good practice anyways, since some of these environments mind have some major changes to what's installed.

Thanks for having a look!

**Pull request merged**